pkg/changelog/fake_impl.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package changelog
+
+import (
+	"k8s.io/release/pkg/changelog/changelogfakes"
+)
```

Comment:
I finally got some time to understand what is going on here. 

Somehow, having `NewFakeImpl()` as a function of the changelog package doesn't ring right. Not only it is not _really_ creating a new Fake (all its fields are overridden with the real `defaultImpl` methods) but its mainly more of a temporary maneuver around the URL change.

If we do it this way, instead of introducing a new file and a new package-level function, I would simply build the `FakeImpl{}` with the real methods inside `cmd.TestNewMinorRelease()`. But, even then, hardcoding the real methods into the stub fields sounds like the function will break at some point if new methods are introduced.

An idea: If the problem is that old releases need to check `oldURL` to get their MD file, why don't we commit a copy of the md draft in the new directory to `/releases/release-1.17`? We could test using v1.20 which already has the new release-notes directory to alter less stuff. That small, pure markdown commit translates into three main advantages:
1. Cleaner codebase and API surface as we would avoid the `NewFakeImpl()` package-level function
2. Avoid having the somewhat hardcoded maneuver around the 1.17 test URL
3. We avoid introducing the risk of the test failing if new critical methods are introduced into the changelog pkg and we do not add them in `NewFakeImpl`

I know that having a copy of the markdown does seem too hygienic either but saves us a lot of trouble in the go code.

Reply:
so yeah I tried going down that approach, on top of the noise which you mentioned, the whole test needs to be changed drastically -- particularly on this line which we rely to determine the passing of the test: https://github.com/kubernetes/release/blob/50d01dfab03b285da655932771b60b9bc109a601/cmd/krel/cmd/changelog_test.go#L199

I agree that having the `NewFakeImpl` looks off -- would it change the view if we name it as `NewModifiableImpl`?

alternatively, I'm willing to do a rewrite on these tests to be simpler and more intentional on what they do, but that would be something bigger than this PR 😄 I can own that effort to move forward if you feel okay with revisiting the way we test things on this section of the codebase.